### ISSUE TITLE
Supporting headings

### ISSUE THREAD

**Issue by crazymedguy** [tadashi-aikawa/obsidian-various-complements-plugin#72]
I use a lot of heading embeds/refs, which can be triggered by `[[##`. I think it can give a lot of extension to just pages `[[`. Would be nice if we could link to existing headings.

**Comment by tadashi-aikawa** [comment]
Hi, @crazymedguy. Thank you for your FR :)

I don't use complete headers, but I have understood what and why you want to do. I'll implement it 👍

**Comment by firinael** [comment]
Bumping this since I downloaded the plugin literally because I thought it could do this lol.

**Comment by tadashi-aikawa** [comment]
#167

**Comment by Andrea-28-03** [comment]
Hi, can you implement this, please it could be really usefull and change how i organize my vault, making everything more simple

**Comment by tadashi-aikawa** [comment]
@Andrea-28-03 
I plan to take on the challenge soon. 👍

**Comment by usernotnull** [comment]
+1

**Comment by Nulllink** [comment]
Yes, want to see this feature too. Even thinking to try and make it by myself🤪

**Comment by tadashi-aikawa** [comment]
This could be a good new feature for the anniversary Various Complements version 10 🤔 

(I think performance degression is inevitable...)

**Comment by Septikai** [comment]
Also bumping this because I've been looking for something that could do this and integrating it into this plugin would be really nice, if performance degression is an issue then it could always be a setting to toggle (assuming that isnt too difficult to do).

**Comment by Nulllink** [comment]
❤

**Comment by PedsaTime** [comment]
Great! I was about to make a post asking for this!

**Comment by SiddanthEmani** [comment]
Voting to add this feature soon :)

**Comment by jques17** [comment]
Also adding my support to this feature request. I would also share my current workaround and a possible solution:

### Workaround
My current workaround is adding the headers of a note as an alias in YAML. That way, I can easily flow into typing, but I do have to add '#Header' to the link created. For example, I want to link to a note name 'Numbers', with a Header named 'twenty-three':

1. I add 'twenty-three' as an alias
2. When I type twenty-three, various complements gives me [[Numbers|twenty-three]]
3. I then modify [[Numbers|twenty-three]] to [[Numbers#twenty-three|twenty-three]].

### Proposed solution
My proposed solution would be to add the headers in the alias with a special character, for example an "@!". Various complements would then detect that special character, and recognize it as a special alias that needs header replacement. In my previous example, typing "@!twenty-three" should result to [[Numbers#twenty-three|twenty-three]]. The downside here is that you will need to type the headers in the alias field of YAML.

**Comment by MelecaZane** [comment]
Downloaded the plugin purely because I thought it could do this!

**Comment by sqlerrorthing** [comment]
+1
